recipes/duktape/all/conanfile.py

```diff
@@ -0,0 +1,61 @@
+import os
+from conans import CMake, ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+from conans.tools import Version
+
+
+class DuktapeConan(ConanFile):
+    name = "duktape"
+    license = "MIT"
+    description = "Duktape is an embeddable Javascript engine, with a focus on portability and compact footprint."
+    topics = ("javascript", "engine", "embeddable", "compact")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://duktape.org"
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.configure()
+        return self._cmake
+
+    def build(self):
+        cmake = self._configure_cmake()
```

Comment:
https://duktape.org/guide.html#compiling
By running `tools/configure.py`, you use duktape as it is supposed to be + options can be added.

Reply:
can you make a proposal? I don't know conan enough to use this python script inside conan....

is there something like `CMake` python class to call the python script?

Reply:
Nope, conan has no special handlers for python.
But it has support for creating subprocesses (such as the one used in test_package).

Something like:
```python
self.run("python tools/configure.py --output-directory \"{}\" -DDUK_USE_FASTINT -UDUK_USE_ES6_PROXY".format(self.build_folder), run_environment=True)
```
You can add options by doing:
```python
conf_opt = lambda opt, b: ("-D" if opt else "-U") + opt
conf_args = [
    conf_opt("DUK_USE_FASTINT, self.options.use_fastint),
    conf_opt("DUK_USE_ES6_PROXY, self.with_es6_proxy),
]
self.run("python tools/configure.py --output-directory \"{}\" {}".format(self.build_folder, " ".join(conf_args), run_environment=True)
```
I don't know the complete list of options, but https://github.com/svaarala/duktape/blob/master/doc/duk-config.rst is a good starting point.
I also think one will have to grep the duktape sources to find the defines that can be configured.
Notice also that the configure script has a `--dll` option for creating a shared library.